src/Reflection/ReflectionMethod.php

```diff
@@ -30,6 +30,8 @@ class ReflectionMethod extends ReflectionFunctionAbstract
 
     private ReflectionClass $implementingClass;
 
+    private ReflectionClass $currentClass;
```

Comment:
I am a bit confused here: why are there 3 classes?

 * `$declaringClass` - obviously the topmost level
 * `$implementingClass` - potentially the level at which the concrete method is introduced (if no override occurred)
 * `$currentClass` - the class where we currently are, ignoring inheritance?

We could need a line or two explaining that :|

Reply:
I've added some comments to the methods...